internal/exec/valdiate.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/coreos/ignition/config"
+)
+
+func Validate(filename string) error {
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	_, r, err := config.Parse(b)
+	fmt.Printf("Validation report:\n%v\n", r)
+	if err != nil {
+		return err
+	}
+
+	if r.IsFatal() {
+		fmt.Println("Config had fatal errors")
+	} else {
+		fmt.Println("Config is valid")
```

Comment:
```
Validation report:
warning at line 3, column 8
    2:   "ignition": {"version": "2.0.0"},
    3:   "foo"
             ^
Config has unrecognized key: foo
Config is valid
```

which is not exactly right.

Also, in general, Unix utilities say nothing on success.  That is arguably not very user-friendly, depending on the user's expectations, but it does ease use in scripts.

Reply:
There are different level's of validity. A config can have unrecognized keys (or other warnings) and still be "valid".

The "config is valid" line is to clarify that, in the case of warnings, the config is still valid.